### ISSUE TITLE
Using sys_calloc results in "undefined reference to `_malloc'" linker error

### ISSUE THREAD

**Issue by Heath123** [Jonimoose/libfxcg#75]
Compiling this:
```c
#include <fxcg/heap.h>

int main(void) {
  // Use calloc to allocate memory
  char *str = sys_calloc(10, sizeof(char));
}
```
results in this:
```
sh3eb-elf-gcc -MMD -MP -MF /home/heath/PrizmSDK-win-0.5.2/projects/calloc-repro/build/example.d -Wall -Wno-unused-label -Wno-unused-value -mb -m4a-nofpu -mhitachi -nostdlib   -I/home/heath/PrizmSDK-win-0.5.2/projects/calloc-repro/build -I/home/heath/PrizmSDK-win-0.5.2/include -ffunction-sections -fdata-sections -c /home/heath/PrizmSDK-win-0.5.2/projects/calloc-repro/src/example.c -o example.o 
/home/heath/PrizmSDK-win-0.5.2/projects/calloc-repro/src/example.c: In function 'main':
/home/heath/PrizmSDK-win-0.5.2/projects/calloc-repro/src/example.c:5:9: warning: unused variable 'str' [-Wunused-variable]
    5 |   char *str = sys_calloc(10, sizeof(char));
      |         ^~~
sh3eb-elf-gcc  example.o -mb -m4a-nofpu -mhitachi -nostdlib -T/home/heath/PrizmSDK-win-0.5.2/toolchain/prizm.x -Wl,-static -Wl,-gc-sections  -L/home/heath/PrizmSDK-win-0.5.2/lib -lc -lfxcg -lgcc -o /home/heath/PrizmSDK-win-0.5.2/projects/calloc-repro/calloc-repro.bin
/usr/local/cross/lib/gcc/sh3eb-elf/11.2.0/../../../../sh3eb-elf/bin/ld: /tmp/cc2dN5ey.ltrans0.ltrans.o: in function `_sys_calloc':
<artificial>:(.text+0x28): undefined reference to `_malloc'
collect2: error: ld returned 1 exit status
make[1]: *** [/home/heath/PrizmSDK-win-0.5.2/toolchain/prizm_rules:83: /home/heath/PrizmSDK-win-0.5.2/projects/calloc-repro/calloc-repro.bin] Error 1
make: *** [Makefile:108: all] Error 2
```

**Comment by dr-carlos** [comment]
I have created a PR that fixes this, see #76.

**Comment by Heath123** [comment]
Thanks, it works now
